streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -389,6 +395,10 @@ private void shutdown() {
             log.error("{} Failed to close restore consumer: ", logPrefix, e);
         }
 
+        // hotfix to improve ZK behavior als long as KAFKA-4060 is not fixed (c.f. KAFKA-4369)
+        // when removing this, make StreamPartitionAssignor#internalTopicManager "private" again
+        partitionAssignor.internalTopicManager.zkClient.close();
```

Comment:
Generally speaking I'd suggest not having a single PR involved with multiple fixes, since it will make reviewing much harder. But since KAFKA-4369 will not be fixed this may be OK. Could you add a `TODO` at the beginning to make it more visible?